activestorage/test/models/attachment_test.rb

```diff
@@ -134,6 +134,15 @@ class ActiveStorage::AttachmentTest < ActiveSupport::TestCase
     assert_equal blob, ActiveStorage::Blob.find_signed(signed_id)
   end
 
+  test "can destroy attachment without existing relation" do
+    blob = create_blob
+    @user.highlights.attach(blob)
+    attachment = @user.highlights.find_by(blob_id: blob.id)
+    attachment.update_attribute(:name, "old_highlights")
```

Comment:
Actually, if I comment out this `update_attribute` the test passes pre-patch. I wonder why? 🤔

Reply:
It should pass, since highlights is still defined on the User class. The idea here is that if the attachment is linked via an association that is no longer defined, then the attachment record shouldn't raise an error when deleted. 

Admittedly the developer should probably clean these up in a data-migration or a rake task if that's the intention, but this also shouldn't error either.

---

Do you think there is a better way to test this scenario? I thought it would be easiest to just simply change the association name stored on the attachment record after it is created.

Reply:
@DRSisco Sorry for the possibly half-baked question but in this scenario the `:name` attribute is referencing the association?

Reply:
@zzak Yeah, `:name` is used for the reverse lookup, so by changing name, we can mirror what it would be if the attachment still being attached to a record, but under an association that doesn't exist at this point in the code base.